uw_oidc/middleware.py

```diff
@@ -25,34 +29,35 @@ def process_view(self, request, view_func, view_args, view_kwargs):
 
         if 'HTTP_AUTHORIZATION' in request.META:
             try:
+                if request.user.is_authenticated:
+                    # not validate ID token on every request
+                    return None
+
+                # We are seeing this user for the first time in this
+                # session, attempt to authenticate the user.
                 token = request.META['HTTP_AUTHORIZATION']
                 username = self.clean_username(
                     UWIdPToken().username_from_token(token))
 
-                if request.user.is_authenticated:
-                    if request.user.get_username() != username:
-                        # An authenticated user is associated with the request,
-                        # but it does not match the user in the token.
-                        raise InvalidTokenError('Username mismatch')
-                else:
-                    # We are seeing this user for the first time in this
-                    # session, attempt to authenticate the user.
-                    user = auth.authenticate(request, remote_user=username)
-                    if user:
-                        # User is valid.  Set request.user and persist user
-                        # in the session by logging the user in.
-                        auth.login(request, user)
-                        request.session[self.TOKEN_SESSION_KEY] = token
-
+                user = auth.authenticate(request, remote_user=username)
+                if user:
+                    # User is valid.  Set request.user and persist user
+                    # in the session by logging the user in.
+                    auth.login(request, user)
+                    request.session[self.TOKEN_SESSION_KEY] = token
+                    logger.info(json.dumps(
```

Comment:
I don't see the value of this logging.

Reply:
It is helpful from MyUW's stand point to know when a token based session is initiated.